src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -2419,5 +2439,31 @@ public QueueItem PeekNextRequestForConnectionAttempt()
 
             public int RequestsWithoutAConnectionAttempt => _size - _attemptedConnectionsOffset;
         }
+
+        private sealed class HttpConnectionWaiter<T> : TaskCompletionSourceWithCancellation<T>
```

Comment:
There are no other usages of `TaskCompletionSourceWithCancellation<T>`. Should we merge these types?

Reply:
@greenEkatherine is adding another use in a PR.